src/GlobalForm.tsx

```diff
@@ -1,17 +1,23 @@
-import { activeSubmitButtonState, useCatalog, useFiles, useWorkflow } from './store'
+import { useSetActiveSubmitButton, useCatalog, useFiles, useWorkflow } from './store'
 import { internalizeDataUrls } from './dataurls'
 import { Form } from './Form'
-import { useSetRecoilState } from 'recoil'
+import { groupParameters } from './grouper'
 
 export const GlobalForm = (): JSX.Element => {
   const { global: globalSchemas } = useCatalog()
   const { setGlobalParameters, global: parameters } = useWorkflow()
   const files = useFiles()
-  const submitFormRefSetter = useSetRecoilState(activeSubmitButtonState) as (instance: HTMLButtonElement | null) => void
-  const parametersWithDataUrls = internalizeDataUrls(parameters, files) // TODO move to hook, each time component is re-rendered this method will be called
-  const uiSchema = (globalSchemas.uiSchema != null) ? globalSchemas.uiSchema : {}
+  const submitFormRefSetter = useSetActiveSubmitButton()
+  // TODO move to hook, each time component is re-rendered this method will be called
```

Comment:
This sounds like something you want to do soonish, right?

Reply:
yep, fixed it by using custom hook with a [writable recoil selector](https://recoiljs.org/docs/api-reference/core/selector#writeable-selectors).